src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventValidator.java

```diff
@@ -13,41 +14,97 @@
+        Event referencePeriodStart = eventMap.get(EventService.Tag.ref_period_start.toString());
+        Event referencePeriodEnd = eventMap.get(EventService.Tag.ref_period_end.toString());
 
+        return referencePeriodInCorrectOrder(referencePeriodStart, referencePeriodEnd)
+                && remindersDuringExercise(eventMap);
+    }
 
-        if (!mpsEvent.getTimestamp().before(goLiveEvent.getTimestamp())) {
-            return false;
+    private boolean referencePeriodInCorrectOrder(final Event referencePeriodStart, final Event referencePeriodEnd) {
+        boolean isValid = true;
+        if (referencePeriodStart != null && referencePeriodEnd != null) {
+            isValid = referencePeriodStart.getTimestamp().before(referencePeriodEnd.getTimestamp());
         }
+        return isValid;
+    }
 
-        if (!goLiveEvent.getTimestamp().before(returnByEvent.getTimestamp())) {
-            return false;
-        }
 
-        if (!returnByEvent.getTimestamp().before(exerciseEndEvent.getTimestamp())) {
-            return false;
+    private boolean remindersDuringExercise(final Map<String, Event>eventMap) {
+        Event goLive = eventMap.get(EventService.Tag.go_live.toString());
+        Event exerciseEnd = eventMap.get(EventService.Tag.exercise_end.toString());
+
+        Event reminder1 = eventMap.get(EventService.Tag.reminder_1.toString());
+        Event reminder2 = eventMap.get(EventService.Tag.reminder_2.toString());
+        Event reminder3 = eventMap.get(EventService.Tag.reminder_3.toString());
+
+        return eventDuringExercise(goLive, reminder1, exerciseEnd)
+                && eventDuringExercise(goLive, reminder2, exerciseEnd)
+                && eventDuringExercise(goLive, reminder3, exerciseEnd);
+
+    }
+
+    private boolean eventDuringExercise(Event goLive, Event event, Event exerciseEnd) {
+        boolean isValid = true;
+        if(event != null) {
+            isValid = goLive.getTimestamp().before(event.getTimestamp())
+                    && event.getTimestamp().before(exerciseEnd.getTimestamp());
         }
+        return isValid;
+    }
 
-        return true;
+    /**
+     * Validates the mandatory events are in the following order:
+     * MPS
+     * Go Live
+     * Return By
+     * Exercise End
+     */
+    private boolean validateMandatoryEvents(final Map<String, Event> eventMap) {
+        Event mpsEvent = eventMap.get(EventService.Tag.mps.toString());
+        Event goLiveEvent = eventMap.get(EventService.Tag.go_live.toString());
+        Event returnByEvent = eventMap.get(EventService.Tag.return_by.toString());
+        Event exerciseEndEvent = eventMap.get(EventService.Tag.exercise_end.toString());
+
+        return mpsEvent.getTimestamp().before(goLiveEvent.getTimestamp())
+                && goLiveEvent.getTimestamp().before(returnByEvent.getTimestamp())
+                && returnByEvent.getTimestamp().before(exerciseEndEvent.getTimestamp());
```

Comment:
The check before was !returnByEvent.getTimestamp().before(exerciseEndEvent.getTimestamp())

Reply:
it had return false after that, so if the return byevent wasn't before exercise end it was invalid.

Reply:
ah okay